include/convex_bodies/hpolytope.h

```diff
@@ -638,6 +639,124 @@ class HPolytope {
+                                    VT& Av,
+                                    int& facet_prev) const
+    {
+        NT lamda = 0, lamda2 =0, lamda1 =0, alpha;
+        NT min_plus  = std::numeric_limits<NT>::max();
+        NT max_minus = std::numeric_limits<NT>::lowest();
+        VT sum_nom;
+        bool real;
+        int m = num_of_hyperplanes(), facet = -1;
+
+        Ar.noalias() = A * r.getCoefficients();
+        sum_nom = Ar - b;
+        Av.noalias() = A * v.getCoefficients();;
+
+        NT* Av_data = Av.data();
+        NT* sum_nom_data = sum_nom.data();
+        NT* Ac_data = Ac.data();
+
+        for (int i = 0; i < m; i++) {
+            alpha = -((*Ac_data) / (2.0 * T));
+            solve_quadratic_polynomial(alpha, (*Av_data), (*sum_nom_data), lamda1, lamda2, real);
+            if (real) {
+                lamda = pick_intersection_time(lamda1, lamda2, i, facet_prev);
+                if (lamda < min_plus && lamda > 0) {
+                    min_plus = lamda;
+                    facet = i;
+                }
+            }
+            Av_data++;
+            sum_nom_data++;
+            Ac_data++;
+        }
+        facet_prev = facet;
+        return std::make_pair(min_plus, facet);
+    }
+
+    std::pair<NT, int> quadratic_positive_intersect(Point const& r,
+                                    Point const& v,
+                                    VT& Ac,
+                                    NT const& T,
+                                    VT& Ar,
+                                    VT& Av,
+                                    NT const& lambda_prev,
+                                    int& facet_prev) const
+    {
+
+        NT lamda = 0, lamda2 =0, lamda1 =0, alpha;
+        NT min_plus  = std::numeric_limits<NT>::max();
+        NT max_minus = std::numeric_limits<NT>::lowest();
+        VT sum_nom;
+        NT mult;
+        unsigned int j;
+        int m = num_of_hyperplanes(), facet = -1;
+        bool real;
+
+        Ar.noalias() += ((lambda_prev * lambda_prev) / (-2.0*T)) * Ac + lambda_prev * Av;
+        sum_nom = Ar - b;
+        Av.noalias() = A * v.getCoefficients();
+
+        NT* sum_nom_data = sum_nom.data();
+        NT* Av_data = Av.data();
+        NT* Ac_data = Ac.data();
+
+        for (int i = 0; i < m; i++) {
+            alpha = -((*Ac_data) / (2.0 * T));
+            solve_quadratic_polynomial(alpha, (*Av_data), (*sum_nom_data), lamda1, lamda2, real);
+            if (real) {
+                lamda = pick_intersection_time(lamda1, lamda2, i, facet_prev);
+                if (lamda < min_plus && lamda > 0) {
+                    min_plus = lamda;
+                    facet = i;
+                }
+            }
+            Av_data++;
+            sum_nom_data++;
+            Ac_data++;
+        }
+        facet_prev = facet;
+        return std::make_pair(min_plus, facet);
+    }
+
+    NT pick_intersection_time(NT lamda1, NT lamda2, int current_facet, int previous_facet) const
```

Comment:
this actually computes the closest intersection to starting point right? I found the name of the function a bit confusing. `minimum_intersection`, `shortest_intersection` or something along this way?

Reply:
Thanks. I change the name to `pick_first_intersection_time_with_boundary`